File: solutions/c++/Encoded_message.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
	string x;
	cin>>x;
	int k=0,d;
	d=sqrt(x.size());
	 char arr[d][d];
	
	for(int i=0; i<d; i++){
		for(int j=0; j<d; j++){
			arr[i][j]=x[k];
			k+=1;
		}
	}
	for(int i=d-1; i>=0;i--){
		for(int j=0; j<d; j++){
			cout<<arr[j][i];
		}
	}
	
	
	
	return 0;
}

File: Kattis (1)/kattis/Encoded_Message.py
from math import sqrt


x=input()
d=int(sqrt(len(x)))
print(d)
arr=[][10000]
f=0
for i in range(d):
    for j in range(d):
        arr[i][j]=x[f]
        f+=1
for i in range(d-1,0):
    for j in range(d-1):
        print(x[i][j],end="")
 